src/telemetry/telemetry.bif

```diff
@@ -548,3 +558,36 @@ function Telemetry::__dbl_histogram_sum%(val: opaque of dbl_histogram_metric%):
 		return zeek::make_intrusive<DoubleVal>(0.0);
 		}
 	%}
+
+
+function Telemetry::__collect_metrics%(prefix: string, name: string%): any_vec
+	%{
+	auto metrics = telemetry_mgr->CollectMetrics(sv(prefix), sv(name));
+
+	static auto metrics_vector_type = zeek::id::find_type<VectorType>("any_vec");
```

Comment:
Just wondering, is this lookup really that slow that we need to cache it? Is this a common pattern in Zeek code? Putting a static in here basically forces the compiler to put a mutex and something like a `pthread_once` here, so it's definitely not free.

Reply:
Thanks for pointing this out. I wasn't aware of the thread safety / `-fno-threadsafe-statics` aspect here.

There are (older?) examples where a separate initialization function and static variables are used:
https://github.com/zeek/zeek/blob/master/src/file_analysis/File.cc#L41-L79

Newer code seems to use `static auto` directly in functions quite a bit:
https://github.com/zeek/zeek/blob/master/src/IP.cc#L682-L689
https://github.com/zeek/zeek/blob/master/src/zeek.bif#L2285-L2290

Looking at some generated code for a specific gcc compiler, it doesn't look like the mutex is taken everytime, just for the actual initialization. There's likely still some overhead to check whether initialization happened everytime, but not taking a lock heavy overhead: https://godbolt.org/z/Yzshch6eq

I'll keep it for now, but also up changing it to something else if there are more/other opinions.